### PR TITLE
[Fix #1971] Add `space` style for SpaceInsideStringInterpolation

### LINKED ISSUE #1971: Cop requiring for space around interpolation
There is a cop that forbids adding space around string interpolation #1836

It would be good if there was an opposite cop, which **requires** space around string interpolation:
```ruby
# Good
"some #{ interpolated } string"

# Bad
"some #{interpolated} string"
```

### PR DESCRIPTION
**Pull request by gotrevor**

### CODE REVIEW

File: lib/rubocop/cop/style/space_inside_string_interpolation.rb [hunk 1]
```diff
@@ -5,22 +5,36 @@ module Cop
     module Style
       # This cop checks for whitespace within string interpolations.
       #
-      # Good:
-      #    var = "This is the #{good} example"
+      # @example
+      #   Good:
```
Comment (bbatsov) [inline lib/rubocop/cop/style/space_inside_string_interpolation.rb]: You can convert good and bad into line comments, for consistency with the rest of our code.

File: spec/rubocop/cop/style/space_inside_string_interpolation_spec.rb [hunk 2]
```diff
@@ -2,49 +2,96 @@
...
+    let(:cop_config) { { 'EnforcedStyle' => 'space' } }
+    let(:expected_message) do
+      'Missing space around string interpolation detected.'
     end
 
-    it 'does not register an offense for excess literal spacing' do
-      inspect_source(cop, source)
-      expect(cop.messages).to be_empty
+    context 'for ill-formatted string interpolations' do
+      let(:source) { irregular_source << '"#{var}"' }
```
Comment (bbatsov) [inline spec/rubocop/cop/style/space_inside_string_interpolation_spec.rb]: Mutating the original string is not a good idea IMO. Maybe should should simply have two version of it?

**Comment by bbatsov** [comment]
Apart from my small remarks, the changes look good. :+1:

File: spec/rubocop/cop/style/space_inside_string_interpolation_spec.rb (hunk 2 above)
Comment (gotrevor) [inline spec/rubocop/cop/style/space_inside_string_interpolation_spec.rb]: It's not mutating a string, but adding an invalid source to an array of other (always) invalid sources.  Does your comment still stand?
I could make this DRYer with a shared example, and also avoid the array mutating.  I was a bit ambivalent about that idea.

Comment (bbatsov) [inline spec/rubocop/cop/style/space_inside_string_interpolation_spec.rb]: Yeah, it stands. Basically my issue with this is that it becomes harder to keep track of what's going on with the data. Not that big of an issue here, as `irregular_source` is constantly recreated, but still...

Comment (gotrevor) [inline spec/rubocop/cop/style/space_inside_string_interpolation_spec.rb]: Perhaps you'll like this better: https://github.com/bbatsov/rubocop/commit/ea3e2d90e0a08a3866e1c52c9ca959d18a7d8f02

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**